### DISCUSSION TITLE
How does CubicSupercellTransformation work?

### DISCUSSION THREAD

**Discussion by adosar** [materialsproject/pymatgen#3212] [Q&A]
Hello, I have looked at the source code but I can't understand how `CubicSupercellTransformation` from the `pymatgen.transformations_advanced_transformations` module work.  Can someone explain how we arrive at a cubic supercell starting from a non-cubic unit cell?

**Comment by mkhorton** [comment]
It’s a supercell of the input cell. For any input, it’s always possible to define a supercell with ~approximately orthogonal vectors, it’s just a question of how big the supercell is, and how approximate is acceptable.

It can be useful in a variety of contexts for different modelling software, e.g. some TEM image simulation codes.

What’s important to understand is that even if you have it in this cubic-like setting, the symmetry and primitive cell of the crystal have not changed.

**Reply by adosar** [reply]
![cell](https://github.com/materialsproject/pymatgen/assets/110358278/1032fa42-5426-4142-b237-81015e9b1388)

I have added a picture representing a  unit cell (2D for ease of visualization) and a 3x3 supercell. How this supercell can be made cubic?
